code/modules/mob/living/simple_animal/bot/turtle/turtle_cartridge.dm

```diff
@@ -0,0 +1,280 @@
+/datum/stack_item
+	var/line
+	var/skipping
+	var/condition
+
+/datum/stack_item/cmd_if
+/datum/stack_item/cmd_while
+
+
+/obj/item/disk/turtle_cartridge
+	name = "TSL v1.2 Cartridge"
+	desc = "A cartridge for holding Turtle Scripting Language programs. Can be popped into a TurtleBot."
+
+	var/regex/varname_regex = new("^\[a-z\]+$")
+
+	var/list/lines = list()
+	var/memory_limit = 16 //how many variables they can have set at a time
+	var/storage_limit = 1024 //how many lines of code they can write to this cartridge
```

Comment:
This is the only time you use this var. Doesn't look like you check it anywhere.

Reply:
What var

Reply:
`storage_limit`

When it doubt, try the line I commented on.

Reply:
It doesn't show which one you did but that will be used soon.